server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -125,6 +131,10 @@ public void setTaskCancellationService(TaskCancellationService taskCancellationS
         this.cancellationService.set(taskCancellationService);
     }
 
+    public void setTaskResourceTrackingService(TaskResourceTrackingService taskResourceTrackingService) {
+        this.taskResourceTrackingService.set(taskResourceTrackingService);
+    }
```

Comment:
Looking for motivation here as to why ``TaskResourceTrackingService`` needs to be necessarily contained by the TaskManager? Keeping it independent will provide updates of thread context through and independent (and cleaner) routes instead of being from within TaskManager?

Reply:
Current idea is to delegate requests via task manager at most of the places which otherwise would required addition of a param in their constructor.

Reply:
Major Issue was adding this new dependency in `TransportAction`